spec/index.html

```diff
@@ -9558,21 +9556,41 @@ <h4>Aggregate Algebra</h4>
             <div id="defn_algGroup">
               <b>Definition: Group</b>
             </div>
-            <p>Group evaluates a list of expressions against a solution sequence Ψ, producing a
+            <p>Group evaluates a list of expressions against a solution sequence <var>Ψ</var>, producing a
               partial function from keys to solution sequences.</p>
-            <p>Group(exprlist, Ψ) = { ListEval(exprlist, μ) → [ μ' | μ' in Ψ, ListEval(exprlist, μ)
-              = ListEval(exprlist, μ') ] | μ in Ψ }</p>
+            <p>Group(<var>exprlist</var>, <var>Ψ</var>) = {
+              <a href="#defn_ListEval">ListEval</a>(<var>exprlist</var>, <var>μ</var>)
+              → [ <var>μ'</var> | <var>μ'</var> in <var>Ψ</var> such that
+              <a href="#defn_ListEval">ListEval</a>(<var>exprlist</var>, <var>μ'</var>) and
+              <a href="#defn_ListEval">ListEval</a>(<var>exprlist</var>, <var>μ</var>)
+              are the same ] | <var>μ</var> in <var>Ψ</var> },</p>
+            <p>where two lists <var>L</var> and <var>L'</var> (as produced by
+              the <a href="#defn_ListEval">ListEval</a> function) are considered
+              the same iff they have the same number of elements and, for every
+              position&nbsp;<var>k</var> within the two lists, either of the
+              following two conditions is true:</p>
+            <ul>
+              <li>the element at the <var>k</var>-th position of
+                <var>L</var> is an RDF term; the element at the <var>k</var>-th
+                position of <var>L'</var> is also an RDF term; and these two
+                RDF terms are the <a href="#func-sameTerm">same term</a></li>
+              <li>the element at the <var>k</var>-th position of <var>L</var>
+                is an error, and the element at the <var>k</var>-th position of
+                <var>L'</var> is also an error</li>
+            </ul>
           </div>
           <div class="defn">
-            <p><b>Definition: ListEval</b></p>
+            <p><b>Definition: <span id="defn_ListEval">ListEval</span></b></p>
             <p>ListEval((expr<sub>1</sub>, ..., expr<sub>n</sub>), μ) returns a list
               (e<sub>1</sub>, ..., e<sub>n</sub>), where e<sub>i</sub> = expr<sub>i</sub>(μ) or
               error.</p>
             <p>ListEval retains errors resulting from the evaluation of the list elements.</p>
           </div>
-          <p>Note that, although the result of a ListEval can be an error, and errors may be used
+          <p>Note that, although the result of <a href="#defn_ListEval">ListEval</a>
+            may contain errors, and errors may be used
             to group, solutions containing error values are removed at projection time.</p>
```

Comment:
"projection" seems ambiguous here.

```suggestion
            to group, solutions containing error values are removed at the end of evaluating the group and any aggregation functions.</p>
```
The errors do not appear because the overall grouping expression does not let errors out (it may be used within an expression) which is earlier than "project"

(treat this as an observation more than a necessary change)

Reply:
@afs  Thanks for pointing this out! In fact, I think there are more issues with this sentence. In particular, it talks about "solutions containing error values" which is inaccurate because solutions (i.e., solution mappings) cannot contain error values. However, I considered fixing/improving this sentence as out of scope of this PR, whose purpose is to improve the definition of the Group operator. The only reason why I touched this sentence in this PR was to add the link to the definition of ListEval for which the PR introduces a linkable anchor ID to be used in the improved definition of the Group operator.

Reply:
> out of scope of this PR

Please create a new issue for the part that's out of scope of this PR, so we don't lose track of it.

Reply:
> [@TallTed] Please create a new issue for the part that's out of scope of this PR, so we don't lose track of it.

Done: PR #134